src/basic/cgroup-util.c

```diff
@@ -134,6 +134,15 @@ int cg_read_event(const char *controller, const char *path, const char *event,
         return -ENOENT;
 }
 
+bool cg_ns_supported(void) {
+	static thread_local int enabled = -1;
+
+	if (enabled >= 0)
+		return enabled;
+
+	return (enabled = (access("/proc/self/ns/cgroup", F_OK) == 0));
```

Comment:
Not that it matters, but: not too much of a fan of compressing so much stuff into a single line. I think doing the assignment and returning it could well be to two C lines...